Vault Docs - PKI support for PKCS#12 and JKS formats (#2879)
hashicorp/web-unified-docs: content/vault/v2.x/content/api-docs/secret/pki/index.mdx

  (rather than a relative one).

- `format` `(string: "pem")` - Specifies the format for returned data. Can be
  `pem`, `der`, `pem_bundle`, `pkcs12_bundle`, or `jks_bundle`; defaults to `pem`. 
  If `der`, `pkcs12_bundle` or `jks_bundle` the output is base64 encoded. 
  If `pem_bundle`, the `certificate` field will contain the private key and certificate, 
  concatenated; if the issuing CA is not a Vault-derived self-signed root, this will be
  included as well. If `pkcs12_bundle` or `jks_bundle`, the PKCS#12 archive or Java keystore 
  (JKS) contains the private key, certificate, and CA chain.

  ~> **Note** The `jks_bundle` format is provided only for compatibility with legacy 

>   systems and should be avoided for new usage. Prefer `pkcs12_bundle`.

Reviewer: ```suggestion
  - `pem` - Base64 encoded text data.
  - `der` - Base64 encoded binary data.
  - `pem_bundle` - Private key and certificate concatinated in the `certificate`
    field, along with the issuing CA if it is not a Vault-derived, self-signed
    root.
  - `pkcs12_bundle` - Base64 encoded with the private key, certificate, and CA
    chain in the PKCS#12 archive.
  - `jks_bundle` - Provided for compatibility with legacy systems. Avoid in
    favor of `pkcs12_bundle` whenever possible. Base64 encoded with the private
    key, certificate, and CA chain in the Java keystore (JKS).
```
Style correction: write in active voice, make info easy to scan, avoid "this" as a pronoun, avoid putting information in asides that should be part of the normal text

- `private_key_format` `(string: "der")` - Specifies the format for marshaling
  the private key within the private_key response field. Defaults to `der` which will
  return either base64-encoded DER or PEM-encoded DER, depending on the value of
  `format`. The other option is `pkcs8` which will return the key marshalled as
  PEM-encoded PKCS8. `private-key` is not returned as a separate field when format 

>   is `pkcs12_bundle` or `jks_bundle`.

Reviewer: ```suggestion
  PEM-encoded PKCS8. Vault does not return `private-key` as a separate field
  when the format is `pkcs12_bundle` or `jks_bundle`.
```

~> **Note** that this does not apply to the private key within the certificate

>   field if `format` is specified as either `"pem_bundle"`, `"pkcs12_bundle"` or `"jks_bundle"`.

Reviewer: ```suggestion
  field if you specify `format` as `"pem_bundle"`, `"pkcs12_bundle"`, or `"jks_bundle"`.
```
Style correction: write in active voice, use serial (Oxford) commas

- `pkcs12_encoder` `(string: "modern2026")` - Encoder profile to use for PKCS#12 
  archives when format is set to `pkcs12_bundle`. Valid values are `modern2026` and 
  `modern2023`. Defaults to `modern2026`, which uses the newer PKCS#12 integrity 
  format (PBMAC1). 

- `pkcs12_password` `(string: "changeit")` - Password for encrypting the PKCS#12 
  archive when format is set to `pkcs12_bundle`. If not provided, defaults to 
  `changeit`. It is recommended to use the default password and protect the file 

>   using other means or use a high-entropy password. 

Reviewer: ```suggestion
  archive when format is set to `pkcs12_bundle`. We recommend using a high-entropy
  password or using the default password and protecting the file using other means. 
```
Style correction: write in active voice

- `jks_password` `(string: "changeit")` - Password for encrypting the Java keystore 
  when format is set to `jks_bundle`. If not provided, defaults to `changeit`. It is 
  recommended to use the default password and protect the file using other means or 

>   use a high-entropy password. 

Reviewer: ```suggestion
  when format is set to `jks_bundle`. We recommend using a high-entropy password
  or using the default password and protecting the file using other means.
```

- `jks_private_key_alias` `(string: "1")` - The entry alias in the Java keystore (JKS) when 
  format is set to `jks_bundle` and bundle contains a single PrivateKeyEntry. This 
  field is case-sensitive, but relying on case-only differences for unique aliases 
  is not recommended. Defaults to `1`. This parameter is ignored by endpoints that 
  return TrustedCertificateEntry values (trust stores), and instead entries are 

>   assigned incrementing numeric string aliases starting at `"1"`. 

Reviewer: ```suggestion
  format is set to `jks_bundle` and bundle contains a single `PrivateKeyEntry`.
  The `jks_private_key_alias` field is case-sensitive, but you should avoid using
  case-only differences to determine between unique aliases. Endpoints that
  return `TrustedCertificateEntry` values (trust stores) ignore `jks_private_key_alias`
  in favor of assigning incrementing numeric string aliases starting at `"1"`. 
```
Style correction: write in active voice

   field, which can override the user-requested issuer.

- `format` `(string: "pem")` - Specifies the format for returned data. Can be
  `pem`, `der`, `pem_bundle`, `pkcs12_bundle`, or `jks_bundle`; defaults to `pem`. 
  If `der`, `pkcs12_bundle` or `jks_bundle` the output is base64 encoded. 
  If `pem_bundle`, the `certificate` field will contain the private key and certificate, 
  concatenated; if the issuing CA is not a Vault-derived self-signed root, this will be 
  included as well. If `pkcs12_bundle` or `jks_bundle`, the PKCS#12 archive or Java keystore 
  (JKS) contains the private key, certificate, and CA chain.

  ~> **Note** The `jks_bundle` format is provided only for compatibility with legacy 

>   systems and should be avoided for new usage. Prefer `pkcs12_bundle`.

Reviewer: ```suggestion
  - `pem` - Base64 encoded text data.
  - `der` - Base64 encoded binary data.
  - `pem_bundle` - Private key and certificate concatinated in the `certificate`
    field, along with the issuing CA if it is not a Vault-derived, self-signed
    root.
  - `pkcs12_bundle` - Base64 encoded with the private key, certificate, and CA
    chain in the PKCS#12 archive.
  - `jks_bundle` - Provided for compatibility with legacy systems. Avoid in
    favor of `pkcs12_bundle` whenever possible. Base64 encoded with the private
    key, certificate, and CA chain in the Java keystore (JKS).
```
Style correction: Edited to match pattern in previous suggestion

- `private_key_format` `(string: "der")` - Specifies the format for marshaling
  the private key within the private_key response field. Defaults to `der` which will
  return either base64-encoded DER or PEM-encoded DER, depending on the value of
  `format` (`private_key` is not returned as a separate field when format 
  is `pkcs12_bundle` or `jks_bundle`). The other option is `pkcs8` which will return the
  key marshalled as PEM-encoded PKCS8.

~> **Note** that this does not apply to the private key within the certificate
  field if `format` is specified as either `"pem_bundle"`, `"pkcs12_bundle"` or `"jks_bundle"`.

- `pkcs12_encoder` `(string: "modern2026")` - Encoder profile to use for PKCS#12 
  archives when format is set to `pkcs12_bundle`. Valid values are `modern2026` and 
  `modern2023`. Defaults to `modern2026`, which uses the newer PKCS#12 integrity 
  format (PBMAC1). 

- `pkcs12_password` `(string: "changeit")` - Password for encrypting the PKCS#12 
  archive when format is set to `pkcs12_bundle`. If not provided, defaults to 
  `changeit`. It is recommended to use the default password and protect the file 

>   using other means or use a high-entropy password. 

Reviewer: ```suggestion
  archive when format is set to `pkcs12_bundle`.  We recommend using a high-entropy
  password or using the default password and protecting the file using other means. 
```

- `jks_password` `(string: "changeit")` - Password for encrypting the Java keystore 
  when format is set to `jks_bundle`. If not provided, defaults to `changeit`. It is 
  recommended to use the default password and protect the file using other means or 

>   use a high-entropy password. 

Reviewer: ```suggestion
  when format is set to `jks_bundle`. We recommend using a high-entropy
  password or using the default password and protecting the file using other means. 
```

- `jks_private_key_alias` `(string: "1")` - The entry alias in the Java keystore (JKS) when 
  format is set to `jks_bundle` and bundle contains a single PrivateKeyEntry. This 
  field is case-sensitive, but relying on case-only differences for unique aliases 
  is not recommended. Defaults to `"1"`. This parameter is ignored by endpoints that 
  return TrustedCertificateEntry values (trust stores), and instead entries are 

>   assigned incrementing numeric string aliases starting at `"1"`. 

Reviewer: ```suggestion
  format is set to `jks_bundle` and bundle contains a single `PrivateKeyEntry`.
  The `jks_private_key_alias` field is case-sensitive, but you should avoid using
  case-only differences to determine between unique aliases. Endpoints that
  return `TrustedCertificateEntry` values (trust stores) ignore `jks_private_key_alias`
  in favor of assigning incrementing numeric string aliases starting at `"1"`. 
```
Style correction: Edited to match pattern in previous suggestion

  (rather than a relative one).

- `format` `(string: "pem")` - Specifies the format for returned data. Can be
  `pem`, `der`, `pem_bundle`, `pkcs12_bundle`, or `jks_bundle`; defaults to `pem`. 
  If `der`, `pkcs12_bundle` or `jks_bundle` the output is base64 encoded. 
  If `pem_bundle`, the `certificate` field will contain the certificate and, 
  if the issuing CA is not a Vault-derived self-signed root, it will be 
  concatenated with the certificate. If `pkcs12_bundle` or `jks_bundle`, 
  the PKCS#12 archive or Java keystore (JKS) contains the certificate and CA chain 
  (no private key).

~> **Note** The `jks_bundle` format is provided only for compatibility with legacy 

> systems and should be avoided for new usage. Prefer `pkcs12_bundle`.

Reviewer: ```suggestion
  - `pem` - Base64 encoded text data.
  - `der` - Base64 encoded binary data.
  - `pem_bundle` - Certificate contained in the `certificate` field, along
    with the issuing CA if it is not a Vault-derived, self-signed root.
  - `pkcs12_bundle` - Base64 encoded with certificate and CA chain in the PKCS#12
    archive (no private key).
  - `jks_bundle` - Provided for compatibility with legacy systems. Avoid in
    favor of `pkcs12_bundle` whenever possible. Base64 encoded with the private
    key and CA chain in the Java keystore (JKS) (no private key).
```
Edited to match previous suggestion

- `pkcs12_encoder` `(string: "modern2026")` - Encoder profile to use for PKCS#12 
  archives when format is set to `pkcs12_bundle`. Valid values are `modern2026` and 

>   `modern2023`. Defaults to `modern2026`, which uses the newer PKCS#12 integrity 

Reviewer: ```suggestion
  `modern2023`. The `modern2026` profile uses the newer PKCS#12 integrity 
```

  format (PBMAC1).

- `pkcs12_password` `(string: "changeit")` - Password for encrypting the PKCS#12 
  archive when format is set to `pkcs12_bundle`. If not provided, defaults to 
  `changeit`. It is recommended to use the default password and protect the file 

>   using other means or use a high-entropy password. 

Reviewer: ```suggestion
  archive when format is set to `pkcs12_bundle`. We recommend using a high-entropy
  password or using the default password and protecting the file using other means. 
```
Edited to match previous suggestion

- `jks_password` `(string: "changeit")` - Password for encrypting the Java keystore 
  when format is set to `jks_bundle`. If not provided, defaults to `changeit`. It is 

>   recommended to use the default password and protect the file using other means or 

Reviewer: ```suggestion
  when format is set to `jks_bundle`. We recommend using a high-entropy
  password or using the default password and protecting the file using other means. 
```
Edited to match previous suggestion

- `csr` `(string: <required>)` - Specifies the PEM-encoded CSR.

- `format` `(string: "pem")` - Specifies the format for returned data. Can be
  `pem`, `der`, or `pem_bundle`; defaults to `pem`. If `der`, the output is
  base64 encoded. If `pem_bundle`, the `certificate` field will contain the
  private key and certificate, concatenated; if the issuing CA is not a
  Vault-derived self-signed root, this will be included as well.

> - `private_key_format` `(string: "der")` - Specifies the format for marshaling

Author: Noticed this copy pasta on re-review. Sign endpoints do not return a private key (issue endpoints do)

- `csr` `(string: <required>)` - Specifies the PEM-encoded CSR.

> - `format` `(string: "pem")` - Specifies the format for returned data. Can be

Reviewer: ```suggestion
- `format` `(string: "pem")` - Specifies the format for returned data. Can be:
```

  `pem`, `der`, `pem_bundle`, `pkcs12_bundle`, or `jks_bundle`; defaults to `pem`. 
  If `der`, `pkcs12_bundle` or `jks_bundle` the output is base64 encoded. 
  If `pem_bundle`, the `certificate` field will contain the private key and certificate, 
  concatenated; if the issuing CA is not a Vault-derived self-signed root, this will be 
  included as well. If `pkcs12_bundle` or `jks_bundle`, the PKCS#12 archive or Java keystore 

>   (JKS) contains the certificate and CA chain (no private key).

Reviewer: ```suggestion
  - `pem` - Base64 encoded text data.
  - `der` - Base64 encoded binary data.
  - `pem_bundle` - Private key and certificate concatinated in the `certificate`
    field, along with the issuing CA if it is not a Vault-derived, self-signed
    root.
  - `pkcs12_bundle` - Base64 encoded with certificate and CA chain in the PKCS#12
    archive (no private key).
  - `jks_bundle` - Provided for compatibility with legacy systems. Avoid in
    favor of `pkcs12_bundle` whenever possible. Base64 encoded with the private
    key and CA chain in the Java keystore (JKS) (no private key).
```
Edited to match pattern in previous suggestion

- `private_key_format` `(string: "der")` - Specifies the format for marshaling
  the private key within the `private_key` response field. Defaults to `der` which will
  return either base64-encoded DER or PEM-encoded DER, depending on the value of
  `format` (`private_key` is not returned as a separate field when format 
  is `pkcs12_bundle` or `jks_bundle`). The other option is `pkcs8` which will return 

>   the key marshalled as PEM-encoded PKCS8.

Reviewer: ```suggestion
  the private key within the `private_key` response field. Defaults to `der`
  which returns either base64-encoded DER or PEM-encoded DER, depending on the
  value of `format`. Vault does not return `private_key` as a separate field when format 
  is `pkcs12_bundle` or `jks_bundle`. The other option is `pkcs8` which will return 
  the key marshalled as PEM-encoded PKCS8.
```

~> **Note** that this does not apply to the private key within the certificate
  field if `format` is specified as either `"pem_bundle"`, `"pkcs12_bundle"` or `"jks_bundle"`.

- `pkcs12_encoder` `(string: "modern2026")` - Encoder profile to use for PKCS#12 
  archives when format is set to `pkcs12_bundle`. Valid values are `modern2026` and 
  `modern2023`. Defaults to `modern2026`, which uses the newer PKCS#12 integrity 
  format (PBMAC1).

- `pkcs12_password` `(string: "changeit")` - Password for encrypting the PKCS#12 
  archive when format is set to `pkcs12_bundle`. If not provided, defaults to 
  `changeit`. It is recommended to use the default password and protect the file 
  using other means or use a high-entropy password.

> 

Reviewer: ```suggestion
  archive when format is set to `pkcs12_bundle`. We recommend using a high-entropy
  password or using the default password and protecting the file using other means. 

```

- `jks_password` `(string: "changeit")` - Password for encrypting the Java keystore 
  when format is set to `jks_bundle`. If not provided, defaults to `changeit`. It is 
  recommended to use the default password and protect the file using other means or 

>   use a high-entropy password.

Reviewer: ```suggestion
  when format is set to `jks_bundle`. We recommend using a high-entropy
  password or using the default password and protecting the file using other means. 
```

  (rather than a relative one).

- `format` `(string: "pem")` - Specifies the format for returned data. Can be
  `pem`, `der`, `pem_bundle`, `pkcs12_bundle`, or `jks_bundle`; defaults to `pem`. 
  If `der`, `pkcs12_bundle` or `jks_bundle` the output is base64 encoded. 
  If `pem_bundle`, the `certificate` field will contain the certificate and, 
  if the issuing CA is not a Vault-derived self-signed root, it will be 
  concatenated with the certificate. If `pkcs12_bundle` or `jks_bundle`, 
  the PKCS#12 archive or Java keystore (JKS) contains the certificate and CA chain 
  (no private key).

~> **Note** The `jks_bundle` format is provided only for compatibility with legacy 

> systems and should be avoided for new usage. Prefer `pkcs12_bundle`.

Reviewer: ```suggestion
  - `pem` - Base64 encoded text data.
  - `der` - Base64 encoded binary data.
  - `pem_bundle` - Certificate contained in the `certificate` field, along
    with the issuing CA if it is not a Vault-derived, self-signed root.
  - `pkcs12_bundle` - Base64 encoded with certificate and CA chain in the PKCS#12
    archive (no private key).
  - `jks_bundle` - Provided for compatibility with legacy systems. Avoid in
    favor of `pkcs12_bundle` whenever possible. Base64 encoded with the private
    key and CA chain in the Java keystore (JKS) (no private key).
```
Edited to match previous suggestion

- `pkcs12_encoder` `(string: "modern2026")` - Encoder profile to use for PKCS#12 
  archives when format is set to `pkcs12_bundle`. Valid values are `modern2026` and 
  `modern2023`. Defaults to `modern2026`, which uses the newer PKCS#12 integrity 
  format (PBMAC1).

- `pkcs12_password` `(string: "changeit")` - Password for encrypting the PKCS#12 
  archive when format is set to `pkcs12_bundle`. If not provided, defaults to 
  `changeit`. It is recommended to use the default password and protect the file 

>   using other means or use a high-entropy password. 

Reviewer: ```suggestion
  archive when format is set to `pkcs12_bundle`. We recommend using a high-entropy
  password or using the default password and protecting the file using other means.
```

- `jks_password` `(string: "changeit")` - Password for encrypting the Java keystore 
  when format is set to `jks_bundle`. If not provided, defaults to `changeit`. It is 
  recommended to use the default password and protect the file using other means or 

>   use a high-entropy password.

Reviewer: ```suggestion
  when format is set to `jks_bundle`. We recommend using a high-entropy
  password or using the default password and protecting the file using other means.
```

- `csr` `(string: <required>)` - Specifies the PEM-encoded CSR.

- `format` `(string: "pem")` - Specifies the format for returned data. Can be
  `pem`, `der`, `pem_bundle`, `pkcs12_bundle`, or `jks_bundle`; defaults to `pem`. 
  If `der`, `pkcs12_bundle` or `jks_bundle` the output is base64 encoded. 
  If `pem_bundle`, the `certificate` field will contain the certificate and, 
  if the issuing CA is not a Vault-derived self-signed root, it will be 
  concatenated with the certificate. If `pkcs12_bundle` or `jks_bundle`, 
  the PKCS#12 archive or Java keystore (JKS) contains the certificate and CA chain 
  (no private key).

~> **Note** The `jks_bundle` format is provided only for compatibility with legacy 

> systems and should be avoided for new usage. Prefer `pkcs12_bundle`.

Reviewer: ```suggestion
  - `pem` - Base64 encoded text data.
  - `der` - Base64 encoded binary data.
  - `pem_bundle` - Certificate contained in the `certificate` field, along
    with the issuing CA if it is not a Vault-derived, self-signed root.
  - `pkcs12_bundle` - The PKCS#12 archive contains the certificate and CA
    chain (no private key)
  - `jks_bundle` - Provided for compatibility with legacy systems. Avoid in
    favor of `pkcs12_bundle` whenever possible. The Java keystore (JKS) archive
    contains the certificate and CA chain (no private key).
```
Edited to match previous suggestion

- `pkcs12_encoder` `(string: "modern2026")` - Encoder profile to use for PKCS#12 
  archives when format is set to `pkcs12_bundle`. Valid values are `modern2026` and 
  `modern2023`. Defaults to `modern2026`, which uses the newer PKCS#12 integrity 
  format (PBMAC1).

- `pkcs12_password` `(string: "changeit")` - Password for encrypting the PKCS#12 
  archive when format is set to `pkcs12_bundle`. If not provided, defaults to 
  `changeit`. It is recommended to use the default password and protect the file 

>   using other means or use a high-entropy password. 

Reviewer: ```suggestion
  archive when format is set to `pkcs12_bundle`.  We recommend using a high-entropy
  password or using the default password and protecting the file using other means.
```

- `jks_password` `(string: "changeit")` - Password for encrypting the Java keystore 
  when format is set to `jks_bundle`. If not provided, defaults to `changeit`. It is 
  recommended to use the default password and protect the file using other means or 

>   use a high-entropy password.

Reviewer: ```suggestion
  when format is set to `jks_bundle`.  We recommend using a high-entropy
  password or using the default password and protecting the file using other means.
```

  a relative one).

- `format` `(string: "pem")` - Specifies the format for returned data. Can be
  `pem`, `der`, `pem_bundle`, `pkcs12_bundle`, or `jks_bundle`; defaults to `pem`. 
  If `der`, `pkcs12_bundle` or `jks_bundle` the output is base64 encoded. 
  If `pem_bundle`, the `certificate` field will contain the certificate and, 
  if the issuing CA is not a Vault-derived self-signed root, it will be 
  concatenated with the certificate. If `pkcs12_bundle` or `jks_bundle`, 
  the PKCS#12 archive or Java keystore (JKS) contains the certificate and CA chain 
  without a private key for signed certificates.

  ~> **Note** The `jks_bundle` format is provided only for compatibility with legacy 

>   systems and should be avoided for new usage. Prefer `pkcs12_bundle`.

Reviewer: ```suggestion
  - `pem` - Base64 encoded text data.
  - `der` - Base64 encoded binary data.
  - `pem_bundle` - Certificate contained in the `certificate` field along
     with the issuing CA if it is not a Vault-derived, self-signed
    root.
  - `pkcs12_bundle` - The PKCS#12 archive contains the certificate and CA
    chain (no private key) for signed certificates.
  - `jks_bundle` - Provided for compatibility with legacy systems. Avoid in
    favor of `pkcs12_bundle` whenever possible. The Java keystore (JKS) archive
    contains the certificate and CA chain (no private key) for signed certificates.
```
Edited to match previous suggestion

- `pkcs12_encoder` `(string: "modern2026")` - Encoder profile to use for PKCS#12 
  archives when format is set to `pkcs12_bundle`. Valid values are `modern2026` and 
  `modern2023`. Defaults to `modern2026`, which uses the newer PKCS#12 integrity 
  format (PBMAC1). Only applicable when `format=pkcs12_bundle`.

- `pkcs12_password` `(string: "changeit")` - Password for encrypting the PKCS#12 
  archive when format is set to `pkcs12_bundle`. If not provided, defaults to 
  `changeit`. It is recommended to use the default password and protect the file 

>   using other means or use a high-entropy password.

Reviewer: ```suggestion
  archive when format is set to `pkcs12_bundle`. We recommend using a high-entropy
  password or using the default password and protecting the file using other means. 
```

- `jks_password` `(string: "changeit")` - Password for encrypting the Java keystore 
  when format is set to `jks_bundle`. If not provided, defaults to `changeit`. It is 
  recommended to use the default password and protect the file using other means or 

>   use a high-entropy password.

Reviewer: ```suggestion
  when format is set to `jks_bundle`. We recommend using a high-entropy
  password or using the default password and protecting the file using other means. 
```

  absolute end date (rather than a relative one).

- `format` `(string: "pem")` - Specifies the format for returned data. Can be
  `pem`, `der`, `pem_bundle`, `pkcs12_bundle`, or `jks_bundle`; defaults to `pem`. 
  If `der`, the output is base64 encoded. If `pem_bundle`, the `certificate` field 
  will contain the private key (if exported) and certificate, concatenated; if the 
  issuing CA is not a Vault-derived self-signed root, this will be included as well. 
  If `pkcs12_bundle` or `jks_bundle`, the output is a base64-encoded PKCS#12 archive 
  or Java keystore (JKS) containing the certificate and CA chain. For `type=exported`, 
  the private key is included; for other types (`internal`, `existing`, `kms`), only 

>   the certificate and CA chain are included (trust store).

Reviewer: ```suggestion
  - `pem` - Base64 encoded text data.
  - `der` - Base64 encoded binary data.
  - `pem_bundle` - Private key (if exported) and certificate concatinated in the
    `certificate` field, along with the issuing CA if it is not a Vault-derived,
    self-signed root.
  - `pkcs12_bundle` - Base64 encoded PKCS#12 archive containing the certificate
    and CA chain.  Vault only includes the private key when `type=exported`.
  - `jks_bundle` - Provided for compatibility with legacy systems. Avoid in
    favor of `pkcs12_bundle` whenever possible. Base64 encoded Java keystore
    (JKS) containing the certificate and CA chain. Vault only includes the private
    key when `type=exported`.
```
Edited to match previous suggestion

- `private_key_format` `(string: "der")` - Specifies the format for marshaling
  the private key within the `private_key` response field. Defaults to `der` which will
  return either base64-encoded DER or PEM-encoded DER, depending on the value of
  `format`. The other option is `pkcs8` which will return the key marshalled as
  PEM-encoded PKCS8. `private-key` is not returned as a separate field when format 
  is "pkcs12_bundle" or "jks_bundle".

~> **Note** that this does not apply to the private key within the certificate
  field if `format` is specified as either "pem_bundle", "pkcs12_bundle" or "jks_bundle".

- `pkcs12_encoder` `(string: "modern2026")` - Encoder profile to use for PKCS#12 
  archives when format is set to `pkcs12_bundle`. Valid values are `modern2026` and 
  `modern2023`. Defaults to `modern2026`, which uses the newer PKCS#12 integrity 
  format (PBMAC1).

- `pkcs12_password` `(string: "changeit")` - Password for encrypting the PKCS#12 
  archive when format is set to `pkcs12_bundle`. If not provided, defaults to 
  `changeit`. It is recommended to use the default password and protect the file 

>   using other means or use a high-entropy password.

Reviewer: ```suggestion
  archive when format is set to `pkcs12_bundle`.  We recommend using a high-entropy
  password or using the default password and protecting the file using other means.
```

- `jks_password` `(string: "changeit")` - Password for encrypting the Java keystore 
  when format is set to `jks_bundle`. If not provided, defaults to `changeit`. It is 
  recommended to use the default password and protect the file using other means or 

>   use a high-entropy password. Only applicable when `format=jks_bundle`.

Reviewer: ```suggestion
  when format is set to `jks_bundle`. We recommend using a high-entropy
  password or using the default password and protecting the file using other means.
```

- `jks_private_key_alias` `(string: "1")` - The entry alias in the Java keystore (JKS) when 
  format is set to `jks_bundle`. This field is case-sensitive, but relying on 
  case-only differences for unique aliases is not recommended. Defaults to `1`. 
  For `type=exported`, this is used for the PrivateKeyEntry; for other types, this 
  parameter is ignored and entries are assigned incrementing numeric string aliases 

>   starting at `1`. Only applicable when `format=jks_bundle`.

Reviewer: ```suggestion
  format is set to `jks_bundle`. The `jks_private_key_alias` field is
  case-sensitive, but you should avoid using case-only differences to determine
  between unique aliases. Endpoints with `type=exported` use `jks_private_key_alias`
  for the `PrivateKeyEntry`. All other endpoints ignore `jks_private_key_alias`
  in favor of assigning incrementing numeric string aliases starting at `"1"`.
  Only applicable when `format=jks_bundle`.
```
